src/api/commands/currencyprice/CurrencyPriceRootCommand.ts

```diff
@@ -4,30 +4,50 @@ import { RpcCommandInterface } from '../RpcCommandInterface';
 import { validate, request } from '../../../core/api/Validate';
 import { Logger as LoggerType } from '../../../core/Logger';
 import { Types, Core, Targets } from '../../../constants';
-import { CurrencyPriceAddCommand } from './CurrencyPriceAddCommand';
 import { BaseCommand } from '../BaseCommand';
 import { RpcCommandFactory } from '../../factories/RpcCommandFactory';
 import { Commands } from '../CommandEnumType';
+import { CurrencyPriceService } from '../../services/CurrencyPriceService';
+import { MessageException } from '../../exceptions/MessageException';
 
-export class CurrencyPriceRootCommand extends BaseCommand implements RpcCommandInterface<void> {
+export class CurrencyPriceRootCommand extends BaseCommand implements RpcCommandInterface<any> {
 
     public log: LoggerType;
 
     constructor(
-        @inject(Types.Command) @named(Targets.Command.currencyprice.CurrencyPriceAddCommand) private currencyPriceAddCommand: CurrencyPriceAddCommand,
+        @inject(Types.Service) @named(Targets.Service.CurrencyPriceService) private currencyPriceService: CurrencyPriceService,
         @inject(Types.Core) @named(Core.Logger) public Logger: typeof LoggerType
     ) {
         super(Commands.CURRENCYPRICE_ROOT);
         this.log = new Logger(__filename);
     }
 
+    /**
+     *
+     * data.params[]:
+     * [0]: fromCurrency
+     * [1]: toCurrency
+     * [...]: toCurrency
+     *
+     * description: fromCurrency must be PART for now and toCurrency may be multiple currencies like INR, USD etc..
+     * example: [PART, INR, USD, EUR, GBP, ....]
+     */
     @validate()
     public async execute( @request(RpcRequest) data: RpcRequest, rpcCommandFactory: RpcCommandFactory): Promise<any> {
-        return await this.executeNext(data, rpcCommandFactory);
```

Comment:
missing proper jsdoc param and returns:
     * @param data
     * @param rpcCommandFactory
     * @returns {Promise<Bookshelf.Collection<Something>>}
Also, I can't see what this is actually returning.

Reply:
most ide's can generate the jsdoc for you automatically.